library/objective-c/EnvoyHTTPStreamImpl.m

```diff
@@ -62,6 +62,17 @@
   return NULL;
 }
 
+static void *ios_on_send_window_available(envoy_stream_intel stream_intel, void *context) {
```

Comment:
Why can't we use types in this interface and avoid the cast below?

```suggestion
static void *ios_on_send_window_available(envoy_stream_intel stream_intel, ios_context *context) {
```

Reply:
Because the address of this function is ultimately assigned to a C function pointer in the bridge layer which is agnostic of what actually goes in here. It's how Java (and C++/Python, and Rust) can all pass through something different here to handle their respective callbacks with their own types and state.